stdlib/public/core/CMakeLists.txt

```diff
@@ -230,11 +230,9 @@ set(SWIFTLIB_GYB_SOURCES
 set(GROUP_INFO_JSON_FILE ${CMAKE_CURRENT_SOURCE_DIR}/GroupInfo.json)
 set(swift_core_link_flags "${SWIFT_RUNTIME_SWIFT_LINK_FLAGS}")
 set(swift_core_framework_depends)
-set(swift_core_private_link_libraries)
 set(swift_stdlib_compile_flags "${SWIFT_RUNTIME_SWIFT_COMPILE_FLAGS}")
 if(SWIFT_PRIMARY_VARIANT_SDK IN_LIST SWIFT_APPLE_PLATFORMS)
   list(APPEND swift_core_link_flags "-all_load")
-  list(APPEND swift_core_private_link_libraries icucore)
```

Comment:
@vgorloff, this would try to link `icucore` for all cross-compiled stdlibs from a mac before this pull: would this cause problems for you when cross-compiling from mac to Android or [was it simply removed elsewhere](https://github.com/apple/swift/blob/d8de5bafb092a3a5d692c6dbd701cacff0a95c36/stdlib/cmake/modules/AddSwiftStdlib.cmake#L619)?

Reply:
Here is how configured SwiftBuilder on my side.

```sh
-D SWIFT_ANDROID_${arch.swiftArch}_ICU_UC=${icu.paths.installs}/lib/libicuucswift.so
-D SWIFT_ANDROID_${arch.swiftArch}_ICU_UC_INCLUDE=${icu.paths.sources}/source/common
-D SWIFT_ANDROID_${arch.swiftArch}_ICU_I18N=${icu.paths.installs}/lib/libicui18nswift.so
-D SWIFT_ANDROID_${arch.swiftArch}_ICU_I18N_INCLUDE=${icu.paths.sources}/source/i18n
-D SWIFT_ANDROID_${arch.swiftArch}_ICU_DATA=${icu.paths.installs}/lib/libicudataswift.so
```

Also I can't find library `icucore` in build output on my side.
<img width="340" alt="icu" src="https://user-images.githubusercontent.com/3200080/144494837-60cbfe4b-5642-4f9b-bf16-ded2578ec067.png">

Reply:
Well, it should be moot once the ICU dependency is removed, #40340. 🎉